test/functional/cmdLineTests/criu/criuScript.sh

```diff
@@ -23,22 +23,29 @@
 #
 
 echo "start running script";
+# the expected arguments are:
+# $1 is the TEST_ROOT
+# $2 is the JAVA_COMMAND
+# $3 is the JVM_OPTIONS
+# $4 is the MAINCLASS
+# $5 is the NUM_CHECKPOINT
+# $6 is the KEEP_CHECKPOINT
+
 $2 -XX:+EnableCRIUSupport $3 -cp "$1/criu.jar" $4 $5 >testOutput 2>&1;
-if [ "$5" == "SingleCheckpoint" ] || [ "$5" == "TwoCheckpoints" ] || [ "$5" == "ThreeCheckpoints" ]
-then
-sleep 2;
-criu restore -D ./cpData --shell-job;
-fi
-if [ "$5" == "TwoCheckpoints" ] || [ "$5" == "ThreeCheckpoints" ]
-then
-sleep 2;
-criu restore -D ./cpData --shell-job;
-fi
-if [ "$5" == "ThreeCheckpoints" ]
-then
-sleep 2;
-criu restore -D ./cpData --shell-job;
+
+if [ "$6" != true ]; then
+    declare -A checkpoints_map
+    checkpoints_map=( ["SingleCheckpoint"]=1 ["TwoCheckpoints"]=2 ["ThreeCheckpoints"]=3 )
+    for (( num_checkpoint=${checkpoints_map[$5]}; num_checkpoint >= 1; --num_checkpoint ))
+    do
+        sleep 2;
+        criu restore -D ./cpData --shell-job;
+    done
```

Comment:
I meant to set the checkpoint number directly (not a string), so we can set any number in the future.

```
NUM_CHECKPOINT=$5
for i in $(seq $NUM_CHECKPOINT)
do
        sleep 2
        criu restore -D ./cpData --shell-job
done
```

Reply:
The PR has been updated to take numbers of checkpoints as input, the grinder is hyc 25639. Thanks.